src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueDelegate.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon.core.launchqueue.impl
+
+import akka.actor.ActorRef
+import akka.pattern.ask
+import akka.util.Timeout
+import mesosphere.marathon.core.launchqueue.LaunchQueue
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+import LaunchQueue.QueuedTaskCount
+
+import scala.collection.immutable.Seq
+import scala.concurrent.Await
+import scala.concurrent.duration.{ Deadline, _ }
+import scala.util.control.NonFatal
+
+private[launchqueue] class ActorLaunchQueue(actorRef: ActorRef, rateLimiterRef: ActorRef) extends LaunchQueue {
+  override def list: Seq[QueuedTaskCount] =
+    askQueueActor("list")(ActorLaunchQueue.List).asInstanceOf[Seq[QueuedTaskCount]]
+  override def count(appId: PathId): Int =
```

Comment:
Very dense. If it would be a one liner I like that, but here it is a visual block.